mariadb/rootfs/etc/s6-overlay/s6-rc.d/mariadb-pre/run

```diff
@@ -11,8 +11,8 @@ if ! bashio::fs.directory_exists "${MARIADB_DATA}"; then
 fi
 
 # Save variables
-printf "${MARIADB_DATA}" > /var/run/s6/container_environment/MARIADB_DATA
-printf "${NEW_INSTALL}" > /var/run/s6/container_environment/NEW_INSTALL
+cat <<<"${MARIADB_DATA}" > /var/run/s6/container_environment/MARIADB_DATA
+cat <<<"${NEW_INSTALL}" > /var/run/s6/container_environment/NEW_INSTALL
```

Comment:
There is a very slight difference between the two options, printf doesn't add a newline, whereas cat does. I'd go for `echo -n` to make sure this is equivalent.

```suggestion
echo -n "${MARIADB_DATA}" > /var/run/s6/container_environment/MARIADB_DATA
echo -n "${NEW_INSTALL}" > /var/run/s6/container_environment/NEW_INSTALL
```